components/brave_federated/data_stores/data_store.h

```diff
@@ -12,45 +12,52 @@
 #include "base/files/file_path.h"
 #include "base/gtest_prod_util.h"
 #include "base/sequence_checker.h"
+#include "brave/components/brave_federated/public/interfaces/brave_federated.mojom.h"
 #include "sql/database.h"
 
 namespace brave_federated {
 
-FORWARD_DECLARE_TEST(AdNotificationTimingDataStoreTest,
-                     CheckSchemaColumnExistence);
+using TrainingData = base::flat_map<int, std::vector<mojom::CovariatePtr>>;
+
+struct DataStoreTask {
+  int task_id = 0;
+  const std::string task_name;
+  int max_number_of_records = 0;
+  base::TimeDelta max_retention_days;
+};
 
 class DataStore {
  public:
-  explicit DataStore(const base::FilePath& database_path);
+  explicit DataStore(const DataStoreTask data_store_task,
+                     const base::FilePath& database_path);
+  ~DataStore();
 
   DataStore(const DataStore&) = delete;
   DataStore& operator=(const DataStore&) = delete;
 
-  bool Init(int task_id,
-            const std::string& task_name,
-            int max_number_of_records,
-            int max_retention_days);
-  bool DeleteLogs();
-  void EnforceRetentionPolicy();
+  bool Init();
 
-  virtual ~DataStore();
+  int GetNextTrainingInstanceId();
+  void SaveCovariate(const brave_federated::mojom::Covariate& covariate,
+                     int training_instance_id,
+                     const base::Time created_at);
+  bool AddTrainingInstance(
+      const std::vector<brave_federated::mojom::CovariatePtr>
+          training_instance);
+
+  bool DeleteTrainingData();
+  TrainingData LoadTrainingData();
+  void PurgeTrainingDataAfterExpirationDate();
 
  protected:
   friend class DataStoreTest;
-  friend class AdNotificationTimingDataStoreTest;
-  FRIEND_TEST_ALL_PREFIXES(AdNotificationTimingDataStoreTest,
-                           CheckSchemaColumnExistence);
-
-  sql::Database db_;
-  base::FilePath database_path_;
 
-  std::string task_id_;
-  std::string task_name_;
-  int max_number_of_records_;
-  int max_retention_days_;
+  sql::Database database_;
```

Comment:
Rather than having `friend class DataStoreTest` could we not have a getter which returns this object .